Upstream-Status: Backport

Also fixes problems with older clang versions
like we use in dunfell branch.

Signed-off-by: Maksim Sisov <msisov@igalia.com>
---
From 8fc12d7efe98fe651afbb03ef613ae210b422a7d Mon Sep 17 00:00:00 2001
From: Jose Dapena Paz <jdapena@igalia.com>
Date: Tue, 1 Mar 2022 18:06:38 +0000
Subject: [PATCH] GCC: explicitely move return value of
 SCTHashdanceMetadata::ToValue
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

GCC rejects to do Return Value Optimization in
SCTHashdanceMetadata::ToValue, because the copy constructor is
deleted, and in that scenario RVO is rejected in GCC:
    ../../services/network/sct_auditing/sct_auditing_reporter.cc: In member function ‘base::Value network::SCTAuditingReporter::SCTHashdanceMetadata::ToValue() const’:
    ../../services/network/sct_auditing/sct_auditing_reporter.cc:191:10: error: use of deleted function ‘base::Value::Value(const base::Value&)’
      191 |   return value;
          |          ^~~~~
    In file included from ../../services/network/sct_auditing/sct_auditing_reporter.h:14,
                     from ../../services/network/sct_auditing/sct_auditing_reporter.cc:5:
    ../../base/values.h:254:3: note: declared here
      254 |   Value(const Value&) = delete;
          |   ^~~~~

Bug: 819294
Change-Id: I111e51dd10eee7b909d4ac3c0911aac18a589166
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/3488747
Reviewed-by: Chris Thompson <cthomp@chromium.org>
Commit-Queue: José Dapena Paz <jdapena@igalia.com>
Cr-Commit-Position: refs/heads/main@{#976299}
---
 services/network/sct_auditing/sct_auditing_reporter.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/services/network/sct_auditing/sct_auditing_reporter.cc b/services/network/sct_auditing/sct_auditing_reporter.cc
index 900a05dc81b0b..bec982d487428 100644
--- a/services/network/sct_auditing/sct_auditing_reporter.cc
+++ b/services/network/sct_auditing/sct_auditing_reporter.cc
@@ -194,7 +194,7 @@ base::Value SCTAuditingReporter::SCTHashdanceMetadata::ToValue() const {
       kLogIdKey, base::Base64Encode(base::as_bytes(base::make_span(log_id))));
   value.SetKey(kLogMMDKey, base::TimeDeltaToValue(log_mmd));
   value.SetKey(kCertificateExpiry, base::TimeToValue(certificate_expiry));
-  return value;
+  return std::move(value);
 }
 
 // static
-- 
2.25.1

